### DISCUSSION TITLE
Fix edit urls for all preset pages under `Included Presets`

### DISCUSSION THREAD

**Discussion by RahulGautamSingh** [renovatebot/renovate#25548] [Suggest an Idea]
### Tell us more.

When clicking on the edit-icon on all pages under `IncludedPresets` it redirects to `https://github.com/renovatebot/renovate/edit/main/docs/usage/{page_name}`

So, when I click on the edit icon on [DefaultPresets ](https://docs.renovatebot.com/presets-default/)page I am redirected to an non[-existing page](https://github.com/renovatebot/renovate/edit/main/docs/usage/presets-default.md).

We can change this by adding a `edit_url` to the page's metadata to the page we want it redirected to.
In the above case it will be: https://github.com/renovatebot/renovate/edit/main/lib/config/presets/internal/default.ts

Example:

```md
---
edit_url: '<edit-url>'
---
page content ...
```

Since these pages aren't present in `docs/` folder by default as they are generated in the `tools/docs/presets.ts` file we need to modify the generateFrontMatter function in the file.
Before:
https://github.com/renovatebot/renovate/blob/0db0bf03df36c1bfcf41c41b02bd40a60fd74983/tools/docs/presets.ts#L13-L21

After:
```ts
function generateFrontMatter(name: string, order: number): string {
  return `---
date: 2017-12-07
title: ${name} Presets
categories:
    - config-presets
type: Document
order: ${order}
edit_url: https://github.com/renovatebot/renovate/edit/main/lib/config/presets/internal/${name.toLowerCase()}.ts
---
```

**Comment by viceice** [comment]
- #25236

**Reply by HonkingGoose** [reply]
Related issue:

- https://github.com/renovatebot/renovate/issues/19309
